lib/engine/round/operating.rb

```diff
@@ -103,6 +106,29 @@ def must_buy_train?
         @current_entity.trains.empty? && route?
       end
 
+      def buyable_trains
+        depot_trains = @depot.depot_trains
+        other_trains = @depot.other_trains(current_entity)
+
+        # If the corporation cannot buy a train, then it can only buy the cheapest available_on
+        min_depot_train = @depot.min_depot_train
+        if min_depot_train.price > current_entity.cash
+          depot_trains = [min_depot_train]
+
+          if @last_share_sold_price
+            # 1889, a player cannot contribute to buy a train from another corporation
+            return depot_trains unless @ebuy_other_value
+
+            # 18Chesapeake and most others, it's legal to buy trains from other corps until
+            # you've sold more than the face value of the train
+            # so ignore the last share sold in this case
+            available_cash = (current_entity.cash + current_entity.owner.cash) - @last_share_sold_price
```

Comment:
this doesn't make sense to me

if the train costs $100, and combined you have 75$, you sell and share and now have $150, you should still be able to buy the train for $100. but if you ignore the last share, doesn't that mean you can't buy it?

Reply:
i did something similiar in can_sell

`        # Can only sell as much as you need to afford the train
        player = bundle.owner
        total_cash = bundle.price + player.cash + @current_entity.cash
        return false if total_cash >= @depot.min_depot_price + bundle.price_per_share
`

Reply:
> if the train costs $100, and combined you have 75$, you sell and share and now have $150, you should still be able to buy the train for $100. but if you ignore the last share, doesn't that mean you can't buy it?

this is inclusion rather than exclusion. If you had $40 cash, and if the train costs $100 and you've sold a share for $80, you now have $120 cash the $100 train should still be available to buy. I've hopefully improved the comment

You can't use the bundle in this case since the price of the shares change when you've sold them.